### PR TITLE
Add Dashboard object visibility sample and update formatting sample for WorkbookFormattingChanged

### PR DESCRIPTION
**Pull request by b-carver**
Add Dashboard object visibility samples for typescript/javascript and update formatting sample to also listen to WorkbookFormattingChanged event.

### CODE REVIEW

File: Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+<head>
+  <title>Dashboard Object Visibility Sample</title>
+  <!-- jQuery -->
+  <script src="https://code.jquery.com/jquery-3.2.1.min.js"></script>
+
+  <!-- Bootstrap -->
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+
+  <!-- Extensions Library (this will be hosted on a CDN eventually) -->
```
Comment (johnDance) [inline Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.html:12]: Let's remove this line. Thanks

File: Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      const newIsVisible = !currentIsVisible;
+      this.setState(prevState => {
+        const newVisiblityOverrides = new Map(prevState.visiblityOverrides);
+        newVisiblityOverrides.set(id, newIsVisible);
+        return { visiblityOverrides: newVisiblityOverrides };
+      });
+    }
+  }
+
+  console.log('Initializing DashboardObjectVisibility extension.');
```
Comment (johnDance) [inline Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.tsx:88]: Do you want both this log statement and the log statement inside initialize.

File: Samples-Typescript/Formatting/formatting.tsx [hunk 3]
```diff
@@ -1,49 +1,66 @@
...
       const message = 'Tableau formatting information for extensions is missing from this version of Tableau.';
 
-      ReactDOM.render(<>
-        <div style={{color: 'blue'}}>{message}</div>
-      </>, document.getElementById('formattingExample'));
+      return <div style={{ color: 'blue' }}>{message}</div>;
     }
   }
 
   console.log('Initializing Formatting extension.');
```
Comment (johnDance) [inline Samples-Typescript/Formatting/formatting.tsx:64]: (same comment as previous example)

File: Samples/DashboardObjectVisibility/dashboardObjectVisibility.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  <!-- React -->
+  <script src="https://unpkg.com/react@17/umd/react.production.min.js"></script>
+  <script src="https://unpkg.com/react-dom@17/umd/react-dom.production.min.js"></script>
+
+  <!-- Bootstrap -->
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+
+  <!-- Extensions Library (this will be hosted on a CDN eventually) -->
```
Comment (johnDance) [inline Samples/DashboardObjectVisibility/dashboardObjectVisibility.html:16]: remove comment

File: Samples/DashboardObjectVisibility/dashboardObjectVisibility.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      const newIsVisible = !currentIsVisible;
+      this.setState(prevState => {
+        const newVisiblityOverrides = new Map(prevState.visiblityOverrides);
+        newVisiblityOverrides.set(id, newIsVisible);
+        return { visiblityOverrides: newVisiblityOverrides };
+      });
+    }
+  }
+
+  console.log('Initializing DashboardObjectVisibility extension.');
```
Comment (johnDance) [inline Samples/DashboardObjectVisibility/dashboardObjectVisibility.js:72]: same as previous examples

File: Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.html (hunk 1 above)
Comment (b-carver) [reply Samples-Typescript/DashboardObjectVisibility/dashboardObjectVisibility.html]: Removed unnecessary comments and console.log statement (for all comments)

**Review by johnDance — APPROVED** [review]

**Pull request merged**